src/katgpucbf/xbgpu/engine.py

```diff
@@ -966,9 +1149,11 @@ async def start(self, descriptor_interval_s: float = SPEAD_DESCRIPTOR_INTERVAL_S
         # Create the descriptor task first to ensure descriptor will be sent
         # before any data makes its way through the pipeline.
         for pipeline in self._pipelines:
+            # TODO: Make this parametrisable somehow for XPipeline vs BPipeline
+            # BPipeline needs to stagger the sending of descriptors
             descriptor_sender = DescriptorSender(
-                pipeline.send_stream.stream,
-                pipeline.send_stream.descriptor_heap,
+                pipeline.send_stream.stream,  # type: ignore
+                pipeline.send_stream.descriptor_heap,  # type: ignore
```

Comment:
Seems like we should have a base Send class that contains the attributes/functionality that is common between XSend and BSend.

Reply:
I would not be against that, but let's maybe do that via a technical debt ticket once this PR is complete.

Reply:
[NGC-1146](https://skaafrica.atlassian.net/browse/NGC-1146) created accordingly.